src/shared/ui/modal/StackedButtonModal.tsx

```diff
@@ -0,0 +1,89 @@
+import type { useOverlay } from "@/shared/lib";
+import { Button } from "../button";
+import { CloseIcon } from "../icon";
+import { Modal } from "./Modal";
+
+interface StackedButtonModalProps {
+  onClose: ReturnType<typeof useOverlay>["onClose"];
+  children: React.ReactNode;
+  closeIconAriaLabel?: string;
+  title?: string;
+  ConfirmButton?: JSX.Element;
+  CloseButton?: JSX.Element;
+  onConfirm?: () => void | Promise<void>;
+  closeText?: string;
+  confirmText?: string;
+}
+
+const DefaultConfirmButton = ({
+  onClose,
+  onConfirm,
+  confirmText,
+}: Pick<StackedButtonModalProps, "onClose" | "onConfirm" | "confirmText">) => {
+  return (
+    <Button
+      variant="filled"
+      colorType="primary"
+      size="medium"
+      onClick={async () => {
+        onConfirm?.();
+        onClose();
+      }}
+    >
+      {confirmText}
+    </Button>
+  );
+};
+
+const DefaultCloseButton = ({
+  onClose,
+  closeText,
+}: Pick<StackedButtonModalProps, "onClose" | "closeText">) => {
+  return (
+    <Button
+      colorType="tertiary"
+      size="medium"
+      variant="text"
+      type="button"
+      onClick={onClose}
+    >
+      {closeText}
+    </Button>
+  );
+};
+
+export const StackedButtonModal = ({
+  closeIconAriaLabel = "확인창 닫기",
+  title = "",
+  onClose,
+  onConfirm = () => {},
+  children,
+  confirmText = "저장하기",
+  closeText = "취소",
+  ConfirmButton = (
+    <DefaultConfirmButton
+      onClose={onClose}
+      onConfirm={onConfirm}
+      confirmText={confirmText}
+    />
+  ),
+  CloseButton = <DefaultCloseButton onClose={onClose} closeText={closeText} />,
```

Comment:
confirmText나 closeText를 button props에서 정하는 건 어떨까요?
confirm button 컴포넌트를 props로 받는다면, 버튼 안에 들어갈 text도 컴포넌트랑 같이 받으면 좋을 것 같아서요

```suggestion
  ConfirmButton = (
    <DefaultConfirmButton
      onClose={onClose}
      onConfirm={onConfirm}
      confirmText={confirmText}
    >
      저장하기
    </DefaultConfirmButton>
  ),
  CloseButton = <DefaultCloseButton onClose={onClose} closeText={closeText}>취소</DefaultCloseButton>
```

Reply:
`confirmText , onConfirm` 등은 `ConfirmButton` 컴포넌트가 존재하지 않을 때에만 사용하는 `props` 입니다. 

`closeText` 또한 `CloseButton` 을 외부에서 직접 주입하지 않는 경우에만 사용하는 `props` 입니다. 

외부에서 컴포넌트를 직접 주입 하는 경우엔 `confirmText, closeText , onConfirm` 을 넣을 필요가 없습니다. 

해당 영역의 타입을 변경하겠습니다.

Reply:
```tsx
type StackedButton = (onClose: () => Promise<void>) => JSX.Element;

interface StackedButtonModalProps<
  T extends StackedButton | undefined,
  K extends StackedButton | undefined,
> {
  onClose: ReturnType<typeof useOverlay>["onClose"];
  children: React.ReactNode;
  closeIconAriaLabel?: string;
  title?: string;
  ConfirmButton?: T;
  CloseButton?: K;
  onConfirm?: T extends StackedButton ? never : () => void | Promise<void>;
  confirmText?: T extends StackedButton ? never : string;
  closeText?: K extends StackedButton ? never : string;
}
```

다음과 같이 타입을 변경해줬습니다.

외부에서 버튼을 주입 할 땐 `ConfirmText , ..Text , onConfirm` 등을 `props` 로 건내 주면 타입 에러가 발생하게 됩니다.

```tsx
        {ConfirmButton ? (
          ConfirmButton(onClose)
        ) : (
          <DefaultConfirmButton
            onClose={onClose}
            onConfirm={onConfirm}
            confirmText={confirmText || "저장"}
          />
        )}
        {CloseButton ? (
          CloseButton(onClose)
        ) : (
          <DefaultCloseButton
            onClose={onClose}
            closeText={closeText || "취소"}
          />
        )}
```

`StackedButtonModal` 에선 다음과 같이 외부에서 주입 받은 컴포넌트를 사용하게 됩니다.